cloud/terraform/machineactuator.go

```diff
@@ -207,6 +207,10 @@ func (tf *TerraformClient) Create(cluster *clusterv1.Cluster, machine *clusterv1
 		if err != nil {
 			return err
 		}
+		// Save the calico.yaml file.
+		if err := saveFile(podNetworkCNICalico, path.Join("/tmp", "calico.yaml"), 0644); err != nil {
```

Comment:
In the startup script you are letting the user choose which CNI provider to use, in the actuator you are forcing them to save the calico yaml.

What I recommend is de-coupling the CNI from the controller. When we have the machine setup config map in vSphere (probably next week), we could have different startup scripts for different CNI providers. So my suggestion is to install only calico right now (forget about weave), and install it from the startup script.

I also prefer installing from the Calico URL rather than a checked in file, to make upgrading Calico independent from upgrading the machine controller itself. So when there's a new Calico version, we can update the machine setup config map, and not have to change any other machinery. If you have good reason to check it in, I'd be curious to know why (you mentioned you had to change the file -- in what ways?).

One thing I did not ask, what did you do to test that Calico is "WAI"?

Reply:
I did test the network with calico performs correct for pod/service traffic. And calico is configuring the pod with podcider.